test/e2e/network/network_policy.go

```diff
@@ -448,11 +478,12 @@ func createNetworkClientPod(f *framework.Framework, namespace *v1.Namespace, pod
 			Containers: []v1.Container{
 				{
 					Name:  fmt.Sprintf("%s-container", podName),
-					Image: imageutils.GetE2EImage(imageutils.Redis),
+					Image: imageutils.GetBusyBoxImage(),
 					Args: []string{
 						"/bin/sh",
 						"-c",
-						fmt.Sprintf("/usr/bin/printf dummy-data | /bin/nc -w 8 %s.%s %d", targetService.Name, targetService.Namespace, targetPort),
+						fmt.Sprintf("for i in $(seq 1 5); do wget -T 8 %s.%s:%d -O - && exit 0 || sleep 1; done; exit 1",
```

Comment:
again, as long as the pod is ready, and the service has an endpoint, then we shouldn't need to retry; the first attempt should succeed